### PR TITLE
SCI32: Fix QFG4CD Grooper crashes at inn and monolith

### PR DESCRIPTION
**Pull request by sluicebox**
Fixes bugs #10747, #10760

### CODE REVIEW

**Comment by Kawa-oneechan** [comment]
But good _lord_ does QFG4 have a lot of issues :worried:

File: engines/sci/engine/script_patches.cpp [hunk 1]
```diff
@@ -11708,6 +11708,129 @@ static const uint16 qfg4FighterSpearPatch[] = {
...
+	0x81, 0x00,                         // lag 00
+	0x4a, SIG_UINT16(0x0008),           // send 08  [ hero setHeading: 0 self ]
+	0x32, SIG_UINT16(0x00c3),           // jmp 00c3 [ end of method ]
+	SIG_END,
+};
+
+static const uint16 qfg4InnDoorCDPatch[] = {
+	0x31, 0x0f,                         // bnt 0f [ state 2 ]
+	0x38, PATCH_SELECTOR16(normalize),  // pushi normalize
+	0x76,
```
Comment (bluegr) [inline engines/sci/engine/script_patches.cpp]: Please add a push0 comment here

File: engines/sci/engine/script_patches.cpp [hunk 2]
```diff
@@ -11708,6 +11708,129 @@ static const uint16 qfg4FighterSpearPatch[] = {
...
+	0x38, SIG_UINT16(0x0218),           // pushi handsOff [ hard-coded for CD ]
+	SIG_ADDTOOFFSET(+15),
+	0x4a, SIG_UINT16(0x0008),           // send 08 [ hero setStep: ... ]
+	SIG_END,
+};
+
+static const uint16 qfg4WalkDownSlopeCDPatch[] = {
+	0x2f, 0x22,                         // bt 22 [ state 1 ]
+	0x38, PATCH_SELECTOR16(normalize),  // pushi normalize
+	0x76,
```
Comment (bluegr) [inline engines/sci/engine/script_patches.cpp]: Please add a push0 comment here

File: engines/sci/engine/script_patches.cpp [hunk 3]
```diff
@@ -11708,6 +11708,129 @@ static const uint16 qfg4FighterSpearPatch[] = {
...
+	0x38, SIG_UINT16(0x0218),           // pushi handsOff [ hard-coded for CD ]
+	SIG_ADDTOOFFSET(+15),
+	0x4a, SIG_UINT16(0x0008),           // send 08 [ hero setStep: ... ]
+	SIG_END,
+};
+
+static const uint16 qfg4WalkDownSlopeNrsPatch[] = {
+	0x2e, PATCH_UINT16(0x0023),         // bt 0023 [ state 1 ]
+	0x38, PATCH_SELECTOR16(normalize),  // pushi normalize
+	0x76,
```
Comment (bluegr) [inline engines/sci/engine/script_patches.cpp]: Please add a push0 comment here

**Comment by sluicebox** [comment]
Good catch! Way to not be a push0ver.

**Comment by bluegr** [comment]
Thanks! Merging

**Pull request merged**